### PR TITLE
[IMP] base,l10n*: do not require manually designed icons anymore for flags

### PR DESCRIPTION
**Pull request by william-andre**
[task-3166075](https://www.odoo.com/web#id=3166075&model=project.task)

_______

Abandoned commit for now: https://github.com/odoo-dev/odoo/commit/6e1b73fb42c34f92289a281e1f0a4b560040dcc6

### CODE REVIEW

File: odoo/addons/test_lint/tests/test_manifests.py [hunk 1]
```diff
@@ -96,10 +96,16 @@ def _test_manifest_icon_value(self, module, value):
...
+            path_parts = path_string.split('/')
             path = get_module_resource(path_parts[1], *path_parts[2:])
             if not path:
                 _logger.warning(
                     "Icon value specified in manifest of module %s wasn't found in given path."
                     " Please specify a correct value or remove this key from the manifest.",
                     module)
+            if country and not get_module_resource('base', 'static', 'img', 'country_flags', f'{country[0]}.png'):
+                _logger.warning(
+                    "Country value specifiedfor the icon in manifest of module %s wasn't found in given path. "
```
[the remark this answers is not in the document]
Comment (h4818) [reply odoo/addons/test_lint/tests/test_manifests.py:109]: You are using emoji's now, does this still apply?

**Comment by william-andre** [comment]
@stefanorigano What do you think of this?

**Comment by stefanorigano** [comment]
Hello @william-andre , actually we have already redesigned all the flags icons for the next saas :/
The new icons are inline with the new UI... moreover we plan to use inline SVGs to adapt bg and colors according the the color-scheme chosen by the user.
I'm sorry, I wasn't aware about this task, I would have warned you earlier.

cc: @Brieuc-brd 

image

**Comment by stefanorigano** [comment]
Hello @william-andre , actually we have already redesigned all the flags icons for the next saas.
The new icons are inline with the new UI... moreover we plan to use inline SVGs to adapt bg and colors according the the color-scheme chosen by the user.
I'm sorry, I wasn't aware about this task, I would have warned you earlier :/

cc: @Brieuc-brd

image

**Comment by william-andre** [comment]
Previous redesign doesn't prevent this. I'd be happy to change the main icon `addons/account/static/description/l10n.png` if that's the issue.
Like stated on the task, this improves
* remove binaries in the repo
* remove network trafic
* have easy and uniform icons
* possibly customize icons depending on the modules (Invoicing/Stock/Website/...)
* change the design of all l10n icons all at once

I believe that all of these are still valid

Note that your new design doesn't respect the aspect ratio of the flags, this might annoy some people.

**Comment by stefanorigano** [comment]
@william-andre , forget my previous comment, I misled the PR  :)
This is clearly an improvement in terms of both footprint and maintenance.

After the merge we'll glad to apply the necessary adaptations to match the new design (eg. default globe icon, shadows, etc..)
Thanks for the heads-up 👍

File: addons/l10n_latam_base/__manifest__.py [hunk 2]
```diff
@@ -3,7 +3,6 @@
     'name': 'LATAM Localization Base',
     'version': '1.0',
     'category': 'Accounting/Localizations',
-    'sequence': 14,
```
Comment (Levizar) [inline addons/l10n_latam_base/__manifest__.py:6]: ?

**Review by Levizar — COMMENTED** [review]
partial one

File: odoo/tools/misc.py [hunk 3]
```diff
@@ -1729,3 +1729,10 @@ def has_list_types(values, types):
         isinstance(values, (list, tuple)) and len(values) == len(types)
         and all(isinstance(item, type_) for item, type_ in zip(values, types))
     )
+
+def get_flag(country_code: str) -> str:
+    """Get the emoji representing the flag linked to the country code.
+
+    This emoji is composed of the two regional indicator emoji of the country code.
+    """
+    return "".join(chr(int(f"1f1{ord(c)+165:02x}", base=16)) for c in country_code)
```
[the remark this answers is not in the document]
Comment (william-andre) [reply odoo/tools/misc.py:1738]: No, it's the representation of the regional letter. You need two of them to form the flag (flag for `BE` is the combination of the letters `B` and `E`)

File: odoo/addons/base/views/ir_module_views.xml [hunk 4]
```diff
@@ -171,7 +172,15 @@
                     <t t-name="kanban-box">
                       <div class="oe_module_vignette">
                         <t t-set="installed" t-value="record.state.raw_value == 'installed'"/>
-                        <img t-attf-src="#{record.icon.value}" class="oe_module_icon" alt="Icon"/>
+                        <t t-if="record.icon_flag">
+                            <div class="oe_module_icon">
+                                <img t-attf-src="#{record.icon.value.split(';')[0]}" class="w-100" alt="Icon"/>
```
[the remark this answers is not in the document]
Comment (william-andre) [reply odoo/addons/base/views/ir_module_views.xml:177]: Older implementation :see_no_evil:

**Review by Levizar — APPROVED** [review]
@robodoo r+

**Review by Levizar — APPROVED** [review]
@robodoo r+

**Comment by andg-odoo** [comment]
Note that in module info view the flag icon is missing so all l10n modules now look the same: 
image

**Comment by Levizar** [comment]
@andg-odoo I'm not sure to understand:
image

**Comment by Levizar** [comment]
mybad I get it

**Comment by andg-odoo** [comment]
> Mybad I get it: the form view We will make a second pr

Yeah, definitely don't _not_ merge this pr because of the form view, just wanted to let you know for the future.

**Pull request closed without merging**